app/controllers/flow_controller.rb

```diff
@@ -27,11 +30,18 @@ def index
         when 'raw'
           response = order.attributes
         when 'auth'
-          response = order.flow_cc_authorization
-          response = response.success? ? response.params['response'].to_hash : response.message
+          flow_response = order.flow_cc_authorization
+          response      = flow_response.success? ? flow_response.params['response'].to_hash : flow_response.message
         when 'capture'
-          response = order.flow_cc_capture
-          response = response.success? ? response.params['response'].to_hash : response.message
+          flow_response = order.flow_cc_capture
+          response      = flow_response.success? ? flow_response.params['response'].to_hash : flow_response.message
+        when 'refund'
+          response = order.flow_cache['refund']
```

Comment:
once I authorize, capture or refund cc I keep flow api response in cache. is that ok or should I get it live every time?

also that is the way how I track status. if there is "authorize" response hash present, order is authorized.